src/wallet/scriptpubkeyman.cpp

```diff
@@ -246,8 +246,14 @@ bool LegacyScriptPubKeyMan::CheckDecryptionKey(const CKeyingMaterial& master_key
             if (fDecryptionThoroughlyChecked)
                 break;
             else {
-                // Rewrite these encrypted keys with checksums
-                batch.WriteCryptedKey(vchPubKey, vchCryptedSecret, mapKeyMetadata[vchPubKey.GetID()]);
+                // Rewrite these encrypted keys with checksums and signatures
+                uint256 enckey_hash = Hash(vchCryptedSecret);
+                std::vector<unsigned char> sig(64);
+                if (!key.SignSchnorr(enckey_hash, sig, nullptr, uint256())) {
+                    keyFail = true;
+                    break;
+                }
+                batch.WriteCryptedKey(vchPubKey, vchCryptedSecret, mapKeyMetadata[vchPubKey.GetID()], sig);
```

Comment:
Wondering if this change would better fit into the later commit 20451e785f6478b5daf3b06bb2f638bc3662cf9f (_"wallet: Write encrypted key sig if it is not present"_), where the same is (only) done for `DescriptorScriptPubKeyMan`?

Reply:
No, it cannot be moved to later since `WriteCryptedKey` also changes in this commit. The later commit is for DescriptorSPKM only because it doesn't already have an existing upgrading mechanism that needed changes.